sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/ScriptTransformation.scala

```diff
@@ -25,9 +25,18 @@ import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
  * @param input the set of expression that should be passed to the script.
  * @param script the command that should be executed.
  * @param output the attributes that are produced by the script.
+ * @param ioschema the input and output schema applied in the execution of the script.
  */
 case class ScriptTransformation(
     input: Seq[Expression],
     script: String,
     output: Seq[Attribute],
-    child: LogicalPlan) extends UnaryNode
+    child: LogicalPlan,
+    ioschema: Option[ScriptInputOutputSchema]) extends UnaryNode
```

Comment:
What other cases?  There is no need to add complexity to the API unless it
will be used.

In the way you have designed it now, none will result in a confusing error
later in execution.  Just disallow it statically by not making this an
option.
On Feb 1, 2015 11:38 PM, "Liang-Chi Hsieh" <notifications@github.com> wrote:

> In
> sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/ScriptTransformation.scala
> <https://github.com/apache/spark/pull/4014#discussion_r23910289>:
>
> >   */
> >  case class ScriptTransformation(
> >      input: Seq[Expression],
> >      script: String,
> >      output: Seq[Attribute],
> > -    child: LogicalPlan) extends UnaryNode
> > +    child: LogicalPlan,
> > +    ioschema: Option[ScriptInputOutputSchema]) extends UnaryNode
>
> In Hive case, it is not. But I think it may be for other cases?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/4014/files#r23910289>.
>